persistence/mongodb-record/src/main/scala/net/liftweb/mongodb/record/MongoMetaRecord.scala

```diff
@@ -280,6 +292,33 @@ trait MongoMetaRecord[BaseRecord <: MongoRecord[BaseRecord]]
   }
 
   /**
+    * replaces document with new one with given id. if `upsert` is set to true inserts new document
+    * in similar way as save() from sync api
+    */
+  def replaceOneAsync(inst: BaseRecord, upsert: Boolean, concern: WriteConcern) = {
+    useCollAsync { coll =>
+      val p = Promise[BaseRecord]
+      val doc: Document = inst.asDocument
+      val options = new UpdateOptions().upsert(true)
```

Comment:
@marekzebrowski I think this was the only thing left, do you mind if I just go ahead and update the branch?